### PR TITLE
feat(GES): add a new resouce to manage graph

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccGesGraph_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ges -v

=== RUN   TestAccGesGraph_basic
=== PAUSE TestAccGesGraph_basic
=== CONT  TestAccGesGraph_basic
--- PASS: TestAccGesGraph_basic (2645.02s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ges	2645.059s

```

### CODE REVIEW

File: docs/resources/ges_graph.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+* `enable_https` - (Required, Bool, ForceNew) Whether to enable the security mode. This mode may damage GES performance greatly.
+
+  Changing this parameter will create a new resource.
+
+* `arch` - (Optional, String, ForceNew) Graph instance's CPU architecture type.  
+ The value can be **x86_64** or **aarch64**. The default value is **x86_64**.
+
+  Changing this parameter will create a new resource.
+
+* `public_ip` - (Optional, String, ForceNew) The information about public IP.
```
Comment (Jason-Zhang9309) [inline docs/resources/ges_graph.md]:
```suggestion
* `public_ip` - (Optional, List, ForceNew) The information about public IP.  
```

File: docs/resources/ges_graph.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+  The [PublicIp](#GesGraph_PublicIp) structure is documented below.
+
+* `enable_multi_az` - (Optional, Bool, ForceNew) Whether the created graph supports the cross-AZ mode.
+  The default value is false.  
+  If the value is true, the system will create the ECSs in the graph in two AZs.
+
+  Changing this parameter will create a new resource.
+
+* `encryption` - (Optional, String, ForceNew) Whether to enable data encryption The value can be true or false.
```
Comment (Jason-Zhang9309) [inline docs/resources/ges_graph.md]:
```suggestion
* `encryption` - (Optional, List, ForceNew) Whether to enable data encryption The value can be true or false.
```

File: docs/resources/ges_graph.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  The default value is false.  
+
+  Changing this parameter will create a new resource.
+
+  The [Encryption](#GesGraph_Encryption) structure is documented below.
+
+* `lts_operation_trace` - (Optional, String, ForceNew) Whether to enable audit logs. This function is disabled by default.
```
Comment (Jason-Zhang9309) [inline docs/resources/ges_graph.md]:
```suggestion
* `lts_operation_trace` - (Optional, List, ForceNew) Whether to enable audit logs. This function is disabled by default.
```

File: huaweicloud/services/ges/resource_huaweicloud_ges_graph.go [hunk 4]
```diff
@@ -0,0 +1,833 @@
...
+				Description: `The enterprise project ID.`,
+			},
+			"tags": {
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The key/value pairs to associate with the graph.`,
+			},
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ges/resource_huaweicloud_ges_graph.go:145]: "tags": common.TagsForceNewSchema(),

File: docs/resources/ges_graph.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+* `product_type` - (Optional, String, ForceNew) Graph product type.  
+  Value options are as follows:
+    + **InMemory**: memory edition.
+    + **Persistence**: database edition.
+  
+  If **graph_size_type_index** is 6, the value must be **Persistence**.
+
+  Changing this parameter will create a new resource.
+
+* `vertex_id_type` - (Optional, String, ForceNew) ID type of vertices.
```
Comment (Jason-Zhang9309) [inline docs/resources/ges_graph.md]:
```suggestion
* `vertex_id_type` - (Optional, List, ForceNew) ID type of vertices.
```

File: huaweicloud/services/ges/resource_huaweicloud_ges_graph.go [hunk 6]
```diff
@@ -0,0 +1,833 @@
...
+				Computed:    true,
+				ForceNew:    true,
+				Description: `ID type of vertices. This parameter is mandatory only for database edition graphs.`,
+			},
+			"replication": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"keep_backup": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ges/resource_huaweicloud_ges_graph.go:188]: missing in doc

File: docs/resources/ges_graph.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+  Changing this parameter will create a new resource.
+
+* `crypt_algorithm` - (Required, String, ForceNew) Graph instance cryptography algorithm.  
+  Value options are as follows:
+    + **generalCipher**: Chinese cryptographic algorithm.
+    + **SMcompatible**: Commercial cryptography algorithm (compatible with international ones).
+
+  Changing this parameter will create a new resource.
+
+* `enable_https` - (Required, Bool, ForceNew) Whether to enable the security mode. This mode may damage GES performance greatly.
```
Comment (Jason-Zhang9309) [inline docs/resources/ges_graph.md:76]: how about make this optional, and put it with other bool type parameters

Comment (hmbig2) [reply docs/resources/ges_graph.md:76]: cannot, enable this mode just depends on this one parameter.

File: huaweicloud/services/ges/resource_huaweicloud_ges_graph.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_graph.go:188]: fixed

File: docs/resources/ges_graph.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/ges_graph.md]: ok

File: huaweicloud/services/ges/resource_huaweicloud_ges_graph.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_graph.go:145]: auto generate , so keep it as it is.

File: docs/resources/ges_graph.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `arch` - (Optional, String, ForceNew) Graph instance's CPU architecture type.
```
Comment (ShiChangkuo) [inline docs/resources/ges_graph.md]: it's better to use `cpu_arch`, we use this name in many resources and data sources

File: docs/resources/ges_graph.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+
+* `public_bind_type` - (Optional, String) The bind type of public IP.  
+  The valid value are **auto_assign**, and **bind_existing**.
+
+* `eip_id` - (Optional, String) The EIP ID.  
+
+<a name="GesGraph_Encryption"></a>
+The `Encryption` block supports:
+
+* `enable` - (Optional, Bool) Whether to enable data encryption The value can be true or false.
```
Comment (ShiChangkuo) [inline docs/resources/ges_graph.md]:
```suggestion
* `enable` - (Optional, Bool) Whether to enable data encryption. The value can be true or false.
```

dot is missing

File: docs/resources/ges_graph.md [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+
+  The [PublicIp](#GesGraph_PublicIp) structure is documented below.
+
+* `enable_multi_az` - (Optional, Bool, ForceNew) Whether the created graph supports the cross-AZ mode.
+  The default value is false.  
+  If the value is true, the system will create the ECSs in the graph in two AZs.
+
+  Changing this parameter will create a new resource.
+
+* `encryption` - (Optional, List, ForceNew) Whether to enable data encryption The value can be true or false.
```
Comment (ShiChangkuo) [inline docs/resources/ges_graph.md]: how about use `the configuration of data encryption` or something like this.

File: docs/resources/ges_graph.md [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+  The default value is false.  
+
+  Changing this parameter will create a new resource.
+
+  The [Encryption](#GesGraph_Encryption) structure is documented below.
+
+* `lts_operation_trace` - (Optional, List, ForceNew) Whether to enable audit logs. This function is disabled by default.
```
Comment (ShiChangkuo) [inline docs/resources/ges_graph.md]: ditto

File: docs/resources/ges_graph.md (hunk 8 above)
Comment (hmbig2) [reply docs/resources/ges_graph.md]: done

File: docs/resources/ges_graph.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/ges_graph.md]: done

File: docs/resources/ges_graph.md (hunk 11 above)
Comment (hmbig2) [reply docs/resources/ges_graph.md]: done

**Pull request merged**